flow-client/src/test/frontend/ConnectTests.ts

```diff
@@ -396,5 +395,111 @@ describe('ConnectClient', () => {
         await client.call('FooEndpoint', 'fooMethod', {fooParam: 'foo'});
       });
     });
+
+    describe('login', () => {
+      afterEach(() => fetchMock.restore());
+
+      it('on invalid credential', async () => {
+        fetchMock.post('/login', { redirectUrl: '/login?error' });
+        const result = await login('invalid-username', 'invalid-password');
+        const expectedResult = {
+          error: true,
+          errorTitle: 'Incorrect username or password.',
+          errorMessage: 'Check that you have entered the correct username and password and try again.'
+        };
+
+        expect(fetchMock.calls()).to.have.lengthOf(1);
+        expect(result).to.deep.equal(expectedResult);
+      })
+
+      it('on valid credential', async () => {
```

Comment:
Suggestion for more natural description.
```suggestion
      it('should return a CSRF token on valid credentials', async () => {
```

(btw I'm not 100% sure how this CSRF protection should work in our case. At least some Spring example I saw included the token in the login POST request already along with the username and password. Not sure if we should do it like that too? Now we're just expecting it back in the response, not sending it in the request)

Reply:
done